src/nodes.coffee

```diff
@@ -1121,7 +1124,7 @@ exports.Return = class Return extends Base
   astType: -> 'ReturnStatement'
 
   astProperties: (o) ->
-    argument: @expression?.ast(o) ? null
+    argument: @expression?.ast(o, LEVEL_PAREN) ? null
```

Comment:
So went back through existing calls to `.ast()` for child node AST generation and populated the appropriate `level` arguments

Reply:
Someday I'd like to understand how you know which level to use. That would be good to add in a big comment somewhere if it's not written out already (yeah, I'm too lazy to look 😀 )

Reply:
The levels used for child nodes in AST generation here should basically match the corresponding levels used for those children when compiling to JS

But as far as how those work, I haven't seen a thorough explanation - there's a little general comment and examples where the `LEVEL_*`s are defined towards the bottom of `nodes.coffee`

Reply:
"Try compiling and see if it requires more parentheses or not" in general is how it's used. That's why we sometimes had wrong parsing precedence with little to no issue for a long while:
https://github.com/jashkenas/coffeescript/issues/2199#issuecomment-5348891